src/edu/cmu/cs/sasylf/ast/RuleLike.java

```diff
@@ -303,6 +303,132 @@ public Application checkApplication(Context ctx, List<? extends Fact> inputs, Fa
 		return Facade.App(getRuleAppConstant(), allArgs);
 	}
 
+	/**

```

Comment:
Is this duplicating logic from elsewhere?  THis code looks fragile -- we should probably try to limit it to one copy.